simple_youtube_api/Video.py

```diff
@@ -157,15 +157,21 @@ def set_public_stats_viewable(self, viewable: bool):
         else:
             raise Exception("Not a valid status: " + str(viewable))
 
-    # TODO enforce (YYYY-MM-DDThh:mm:ss.sZ) format
-    def set_publish_at(self, time: str):
+    #@video_status_set
+    def set_publish_at(self, publish_at):
         """ Sets time that video is going to be published at in
         (YYYY-MM-DDThh:mm:ss.sZ) format
         """
-        if type(time) == str:
-            self.publish_at = time
+        if (not isinstance(publish_at, datetime.datetime)):
+            raise Exception("publish_at must be a datetime object")
+        if (self.privacy_status != 'private'):
+            raise Exception("In order to be scheduled, the privacy has to be private")
+        
+        if (publish_at >= datetime.datetime.now() and (publish_at.minute == 0 or publish_at.minute == 30)):
```

Comment:
Does timezone matter for the comparison `publish_at >= datetime.datetime.now()`? Like the publish_at is in utc, but the datetime.datetime.now() might not be. See here: https://docs.python.org/3/library/datetime.html#datetime.datetime.now